src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,808 @@
+         end
+         if G.descr==:GL setfield!(G,:X,GAP.Globals.GL(G.deg,G.ring_iso.codomain))
+         elseif G.descr==:SL setfield!(G,:X,GAP.Globals.SL(G.deg,G.ring_iso.codomain))
+         elseif G.descr==:Sp setfield!(G,:X,GAP.Globals.Sp(G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("GO+") setfield!(G,:X,GAP.Globals.GO(1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("SO+") setfield!(G,:X,GAP.Globals.SO(1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("Omega+") setfield!(G,:X,GAP.Globals.Omega(1,G.deg,Int(order(G.ring))))
+         elseif G.descr==Symbol("GO-") setfield!(G,:X,GAP.Globals.GO(-1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("SO-") setfield!(G,:X,GAP.Globals.SO(-1,G.deg,G.ring_iso.codomain))
+         elseif G.descr==Symbol("Omega-") setfield!(G,:X,GAP.Globals.Omega(-1,G.deg,Int(order(G.ring))))
+         elseif G.descr==:GO setfield!(G,:X,GAP.Globals.GO(0,G.deg,G.ring_iso.codomain))
+         elseif G.descr==:SO setfield!(G,:X,GAP.Globals.SO(0,G.deg,G.ring_iso.codomain))
+         elseif G.descr==:Omega setfield!(G,:X,GAP.Globals.Omega(0,G.deg,Int(order(G.ring))))
+         elseif G.descr==:GU setfield!(G,:X,GAP.Globals.GU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) )))
+         elseif G.descr==:SU setfield!(G,:X,GAP.Globals.SU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) )))
+         end
+      end
+      L = GAP.Globals.GeneratorsOfGroup(getfield(G,:X))
+      setfield!(G,:gens,[MatrixGroupElem(G,G.mat_iso(L[i]),L[i]) for i in 1:length(L)])
+
+   else
+      return getfield(G, sym)
+
+   end
+
+end
+
+# add field :X
+function Base.getproperty(x::MatrixGroupElem, sym::Symbol)
+   if sym === :X && !isdefined(x,:X)
+      if !isdefined(x.parent,:ring_iso)
+         fm = gen_mat_iso(x.parent.deg, x.parent.ring)
+         setfield!(x.parent,:ring_iso,fm.fr)
+         setfield!(x.parent,:mat_iso,fm)
+      end
+      setfield!(x, :X, x.parent.mat_iso(x.elm))
+   elseif sym == :elm && !isdefined(x, :elm)                        # assuming that x.X, hence x.parent.ring_iso, are defined
+      setfield!(x, :elm, x.parent.mat_iso(x.X))
+   end
+   return getfield(x,sym)
+end
+
+Base.IteratorSize(::Type{<:MatrixGroup}) = Base.SizeUnknown()
+
+function Base.iterate(G::MatrixGroup)
+  L=GAP.Globals.Iterator(G.X)
+  if GAP.Globals.IsDoneIterator(L)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(L)
+  return MatrixGroupElem(G, G.mat_iso(i),i), L
+end
+
+function Base.iterate(G::MatrixGroup, state)
+  if GAP.Globals.IsDoneIterator(state)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(state)
+  return MatrixGroupElem(G, G.mat_iso(i),i), state
+end
+
+########################################################################
+#
+# Membership
+#
+########################################################################
+
+# this saves the value of x.X
+# x_gap = x.X if this is already known, x_gap = Nothing otherwise
```

Comment:
It's surprising to use `Nothing` the type as a sentinel here, why not `nothing` ?